subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -3273,21 +3278,33 @@ int bt_mesh_cfg_srv_init(struct bt_mesh_model *model, bool primary)
 	return 0;
 }
 
+const struct bt_mesh_model_cb bt_mesh_cfg_srv_cb = {
+	.init = cfg_srv_init,
+};
+
 static void mod_reset(struct bt_mesh_model *mod, struct bt_mesh_elem *elem,
 		      bool vnd, bool primary, void *user_data)
 {
 	size_t clear_count;
 
 	/* Clear model state that isn't otherwise cleared. E.g. AppKey
 	 * binding and model publication is cleared as a consequence
-	 * of removing all app keys, however model subscription clearing
-	 * must be taken care of here.
+	 * of removing all app keys, however model subscription and user data
+	 * clearing must be taken care of here.
 	 */
 
 	clear_count = mod_sub_list_clear(mod);
 
-	if (IS_ENABLED(CONFIG_BT_SETTINGS) && clear_count) {
-		bt_mesh_store_mod_sub(mod);
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		if (clear_count) {
+			bt_mesh_store_mod_sub(mod);
+		}
+
+		bt_mesh_model_data_store(mod, vnd, NULL, 0);
```

Comment:
If I'm reading the settings module correctly, this'll actually generate 0-length entries for models that don't have any user data. I couldn't find a good way to get around this though, do you have any suggestions, or do you think it's fine @jhedberg?

Reply:
A new flag to `model->flags` to indicate if data was ever stored to or retrieved from flash? Similar to what I've tried to do here for pairing keys: https://github.com/jhedberg/zephyr/commit/7e22ca25bc53a97331876bff97c232dcc20a0333

Reply:
Of course 🤦‍♂ Made the change.